app/app.py

```diff
@@ -0,0 +1,49 @@
+from pyrogram import Client, Filters
+
+post_script = "\n\n\n TranslatorBot via pyrogram"
+
+class TranslationApp:
+    def __init__(self, telegram_client, chat_id):
+        self.telegram_client = telegram_client
+        self.chat_id = chat_id
+        self.consumers = []
+        self.translators = []
+
+    def add_consumers(self, *args):
+        self.consumers.extend(args)
+
+    def add_translators(self, *args):
+        self.translators.extend(args)
+
+    def translate(self, text):
+        for t in self.translators:
+            result = t.translate(text)
+            if result:
+                return result
+        return None
+
+    def send_text(self, text):
+        for consumer in self.consumers:
+            consumer.send_text(text)
+
+    def send_photo(self, file_id, caption=None):
+        for consumer in self.consumers:
+            consumer.send_photo(file_id, caption)
+
+    def run(self):
+        @self.telegram_client.on_message(Filters.chat(self.chat_id) & Filters.photo & Filters.caption)
+        def echoPT(client, message):
+            translation = self.translate(message.caption) + post_script
```

Comment:
if translate returns None here you'll throw a type error. Do a none check here or default to returning some other value.

Reply:
Do you think it makes sense to throw an exception here? It is "erroneous behavior" and while the whole app shouldn't come to a stop the current flow should definitely stop if all the translators fail.

Reply:
Found a better fix without using an exception that further reduces code duplication